parlai/core/torch_generator_agent.py

```diff
@@ -870,7 +876,7 @@ def advance(self, logprobs):
                 eostail = _HypothesisTail(
                     timestep=len(self.outputs) - 1,
                     hypid=hypid,
-                    score=self.scores[hypid],
+                    score=self.all_scores[-1][hypid],
```

Comment:
does this mean we're excluding the probability of generating the EOS token?

Reply:
we want to put the value from all_scores here (copied there at line 863), because self.scores is an active buffer which we will penalize on the next time step. Then here it will be a -inf score in the end (this is bad).